### PR TITLE
Fix constants fs

### PR DESCRIPTION
**Pull request by ChinmayShringi**
## Description

Fixes  - Review and document constants in Lind-WASM
(https://github.com/Lind-Project/lind-wasm/issues/46)
(https://github.com/Lind-Project/lind-wasm/issues/30)

This PR reviews and updates the constants defined in the following files:
- `src/constants/mod.rs`
- `src/constants/fs_constants.rs`
- `src/constants/net_constants.rs`
- `src/constants/sys_constants.rs`
- `src/constants/vmmap_constants.rs`

Key changes:
1. Added comprehensive source references to Linux kernel v6.5 headers
2. Added detailed documentation for each constant group
3. Added explanatory comments for individual constants
4. Identified platform-specific and Lind-specific constants
5. Fixed type inconsistencies (u32/i32) where needed
6. Organized constants into logical groups
7. Added references to POSIX.1-2017 standard

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

- `cargo build`

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have verified all constants against official Linux kernel sources
- [x] I have documented platform-specific considerations
- [x] I have identified and documented Lind-specific constants
- [x] I have cross-referenced constants with POSIX standards where applicable

### CODE REVIEW

File: src/constants/mod.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+// Re-export all constants from their respective modules
+pub use fs_constants::*;
+pub use net_constants::*;
+pub use sys_constants::*;
+pub use vmmap_constants::*;
+
+// ===== Memory Protection Flags =====
+// Source: include/uapi/asm-generic/mman-common.h
+// These flags control memory segment permissions for mmap and mprotect
```
Comment (Yaxuan-w) [inline src/constants/mod.rs]: Would it be better to move those to vmmap_constant.rs?

File: src/constants/mod.rs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+pub const O_ACCMODE: i32 = 0o003;    // Mask for file access modes
+pub const O_RDONLY: i32 = 0o0;       // Open read-only
+pub const O_WRONLY: i32 = 0o1;       // Open write-only
+pub const O_RDWR: i32 = 0o2;         // Open read-write
+pub const O_CREAT: i32 = 0o100;      // Create file if it doesn't exist
+pub const O_EXCL: i32 = 0o200;       // Error if O_CREAT and file exists
+pub const O_TRUNC: i32 = 0o1000;     // Truncate file to zero length
+pub const O_APPEND: i32 = 0o2000;    // Append mode - writes always at end
+
+// ===== File Types =====
```
Comment (Yaxuan-w) [inline src/constants/mod.rs]: I'm a bit confused why we keep those here instead of moving to corresponding files?

File: src/safeposix/cage.rs [hunk 3]
```diff
@@ -1,19 +1,35 @@
...
 
 use super::filesystem::normpath;
-pub use super::syscalls::fs_constants::*;
-pub use super::syscalls::net_constants::*;
-pub use super::syscalls::sys_constants::*;
-pub use super::vmmap::*;
-pub use super::vmmap_constants::*;
 
+// Remove these old imports since we're using the central constants module now
+// pub use super::syscalls::fs_constants::*;
```
Comment (Yaxuan-w) [inline src/safeposix/cage.rs]: Would be better to delete directly

File: src/safeposix/syscalls/mod.rs [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 pub mod sys_calls;
-pub mod sys_constants;
 pub use fs_calls::*;
-pub use fs_constants::*;
 pub use net_calls::*;
-pub use net_constants::*;
 pub use sys_calls::*;
-pub use sys_constants::*;
+
+// Removed constants after centralisation
```
Comment (Yaxuan-w) [inline src/safeposix/syscalls/mod.rs]: Same as before

**Review by Yaxuan-w — COMMENTED** [review]
Awesome! Minor suggestions

File: src/safeposix/syscalls/mod.rs (hunk 4 above)
Comment (ChinmayShringi) [reply src/safeposix/syscalls/mod.rs]: Removed commtented code

File: src/safeposix/cage.rs (hunk 3 above)
Comment (ChinmayShringi) [reply src/safeposix/cage.rs]: Removed commtented code

File: src/constants/mod.rs (hunk 1 above)
Comment (ChinmayShringi) [reply src/constants/mod.rs]: removed them from mod.rs

File: src/constants/mod.rs (hunk 2 above)
Comment (ChinmayShringi) [reply src/constants/mod.rs]: removed them from mod.rs

File: src/constants/vmmap_constants.rs [hunk 5]
```diff
@@ -0,0 +1,55 @@
+//! Virtual Memory Mapping Constants Module
```
Comment (rennergade) [inline src/constants/vmmap_constants.rs:1]: I think we should get rid of this file and move all of this to fs_constants

**Comment by rennergade** [comment]
This is great, love the way you set this up. Can you make the change to combine vmmap constants into fs constants. Then I'd like you to set up this PR to be into the monorepo since we're going to deprecate this repository shortly.

**Comment by Yaxuan-w** [comment]
That would be great to move this PR into our new mono-repo :D

**Comment by ChinmayShringi** [comment]
Closing this PR since changes were ported on the monorepo

**Pull request closed without merging**